R/tm_a_pca.R

```diff
@@ -240,7 +247,7 @@ ui_a_pca <- function(id, ...) {
               choices = args$plot_choices,
               selected = args$plot_choices[1]
             ),
-            ui_transform_teal_data(ns("decorate"), transformators = args$decorators)
+            ui_decorate_teal_data(ns("decorator"), decorators = subset_decorators("plot", args$decorators))
```

Comment:
@averissimo so now the pca example should have named list `list(plot = caption_decorator("I am a PCA", "plot"))` as a decorator? I think the current example from the opening of this PR does not have the name in the list in decorator argument

Reply:
Thanks for the comment, this makes me think that for a single output or combination of outputs that always exist, we should only have the `"default"` name.

I was mistaken and we need to disconnect the naming of decorators from the objects, as there is not a 1-1 match.

The 1-1 match is between `{ui, srv}_decorate_teal_data(...)` calls and names.

Reply:
> I was mistaken and we need to disconnect the naming of decorators from the objects, as there is not a 1-1 match.

I think you are referring to the case of `tm_a_regression` where we have 6 objects (plots) but all are named (plot).
I think this is ok to have them all be named `plot` and keep the connection of the named decorator and the name from the internals of the source code.

Reply:
Unsure if this commit was needed https://github.com/insightsengineering/teal.modules.general/pull/812/commits/b29e8d1738f65e4634e40267e672bd4bd9c31958

Reply:
Done in 943c260f

Reply:
Thanks, well done!